### PR TITLE
correctly update form on ajax

### PR DESCRIPTION
**Pull request by mkangia**
## Summary
Found this while working on another ticket where i am trying to remove things related to `ENABLE_DRACONIAN_SECURITY_FEATURES`

In JS file, currently we search for `form.find('#user-password')` which is not present. `user-password` container is actually above the form itself and hence the update does not work. To replicate just add a simple password on a domain that needs strong password.
I also tried replacing the complete form but that lead to issues with CSRF token with the "Cookies must be enabled" page shown. 
So I moved the form tag and the csrf token setup in the main view itself while just the form fields in the partial. 
The partial is also used to render response to the ajax request that submits the form.
Then added a container for the form fields to update them correctly.

I also removed the redundant alert message in case the draconian setting is on (used only by ICDS) which is not needed anymore since the form itself has the error message, which is useful as well for the ticket I was originally working on.

I think this UI can be further improved like the default message about the suggested password does not go away even if you edit the password and also now shows in case the password is not accepted. This happens when we create a new mobile worker correctly.
But this at least fixes the form replacement which is better than how it currently stands

## Product Description
This is only for projects which have "Require Strong Passwords for Mobile Workers" set.
In case the password entered is not acceptable, now user would see a error message after submit on the form itself along with the alert that said "Password was not changed ". I think we can get rid of the alert entirely now. 
Earlier, there was no message shown on the form.
In case the setting `ENABLE_DRACONIAN_SECURITY_FEATURES` is set to true, which is only on ICDS, users won't see the extra message about password in the red alert message anymore but would see it on the form itself.

## Safety Assurance

- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I am certain that this PR will not introduce a regression for the reasons below

### QA Plan

Would pass this through QA considering its a sensitive area. Passwords should continue to meet requirements as set by the project and the environment. 
https://dimagi-dev.atlassian.net/browse/QA-2138

### Safety story
I tested this locally to get the correct error message and then successfully update password after that.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations 

Before
Screenshot from 2020-11-15 17-49-50

After
Screenshot from 2020-11-15 17-48-46

### CODE REVIEW

File: corehq/apps/users/templates/users/partials/reset_password.html [hunk 1]
```diff
@@ -1,5 +1,2 @@
 {% load crispy_forms_tags %}
-<form id="reset-password-form" class="form-horizontal" action="{% url "change_password" domain couch_user.user_id %}" method="post">
-  {% csrf_token %}
-  {% crispy reset_password_form %}
-</form>
+{% crispy reset_password_form %}
```
Comment (orangejenny) [inline corehq/apps/users/templates/users/partials/reset_password.html:2]: Might as well get rid of this file now that it's just two lines.

**Review by orangejenny — APPROVED** [review]
Looks fine, pending QA.

File: corehq/apps/users/templates/users/partials/reset_password.html (hunk 1 above)
Comment (mkangia) [reply corehq/apps/users/templates/users/partials/reset_password.html:2]: Its used as a template for rendering ajax HTML response, so we still need it?

Comment (orangejenny) [reply corehq/apps/users/templates/users/partials/reset_password.html:2]: aha, nevermind then

Comment (esoergel) [reply corehq/apps/users/templates/users/partials/reset_password.html:2]: FWIW, crispy forms can also render in python without a template:
```python
from crispy_forms.utils import render_crispy_form
render_crispy_form(form, context=context)
```

Comment (mkangia) [reply corehq/apps/users/templates/users/partials/reset_password.html]: Sweet.

Danny has not shared much about why this template was ever present in https://github.com/dimagi/commcare-hq/commit/6dfa849e2f210382454db11b86f799a15a89c6db, but does look safe to remove.

https://github.com/dimagi/commcare-hq/pull/28787/commits/78d08aead05e6a83f74122d677aa2052237e29d8

**Comment by mkangia** [comment]
QA done. Merging this now.

**Pull request merged**